lib/tasks/autotest.rake

```diff
@@ -93,8 +93,9 @@ class AutotestSetup
   end
 
   def create_criteria
-    FlexibleCriterion.find_or_create_by(
+    @assignment.criteria.find_or_create_by(
       name: 'criterion',
+      type: 'FlexibleCriterion',
       assessment_id: @assignment.id,
```

Comment:
The change on line 96 to `@assignment.criteria` is good, and it means you can remove this line, since `assessment_id` will be set automatically.